doc/design.rst

```diff
@@ -0,0 +1,23 @@
+.. _design_document:
+
+:octicon:`repo` Design Document
```

Comment:
I propose to either add something meaningful here or to remove the file until an appropriate ticket/pr targets to create a design document.